timeline/views.py

```diff
@@ -52,4 +52,14 @@ def delete(self, request, *args, **kwargs):
         timeline.delete()
         return Response({"message": "Timeline deleted successfully"}, status=status.HTTP_204_NO_CONTENT)
 
+class TimelineList(generics.ListAPIView):
+    serializer_class = TimelineSerializer
+
+    def get_queryset(self):
+        event_id = self.kwargs.get('event_id')  # Assuming event_id is passed as a URL parameter
+
+        queryset = Timeline.objects.filter(task_step__task__event_id=event_id)
+
+        return queryset
     
+
```

Comment:
I think you should add permission classes for the owner event only, this can be done by using accessing the property of task.event.user. This way only the owner of the event could check the timeline and prevent forbidden endpoint.